internal/api/commitment.go

```diff
@@ -2357,10 +2345,19 @@ func liquidProjectMetadataFromDBProject(dbProject db.Project, domain db.Domain,
 	return Some(core.KeystoneProjectFromDB(dbProject, core.KeystoneDomain{UUID: domain.UUID, Name: domain.Name}).ForLiquid())
 }
 
-func evaluateRetryHeader(response liquid.CommitmentChangeResponse, w http.ResponseWriter) {
-	if retryAt, exists := response.RetryAt.Unpack(); exists && response.RejectionReason != "" {
+func commitmentChangeRequestWasRejected(response liquid.CommitmentChangeResponse, w http.ResponseWriter, withHTTPResponse bool) bool {
+	rejected := response.RejectionReason != ""
+	if retryAt, exists := response.RetryAt.Unpack(); exists && rejected {
 		w.Header().Set("Retry-After", retryAt.Format(time.RFC1123))
 	}
+	if rejected && withHTTPResponse {
+		http.Error(w, response.RejectionReason, http.StatusConflict)
+		return true
+	}
+	if rejected {
+		return true
+	}
+	return false
```

Comment:
```suggestion
	return rejected
```
Alternatively, the function can be rewritten to return early from `response.RejectionReason == ""` since everything else is only relevant when it isn't.